src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -1210,17 +1210,32 @@ def _validate_ssh_key_helper(ssh_key_value, should_generate_ssh_keys):
     return content
 
 
-def _validate_vm_vmss_msi(cmd, namespace, from_set_command=False):
-    if from_set_command or namespace.assign_identity is not None:
+def _validate_vm_vmss_msi(cmd, namespace, is_identity_assign=False):
+
+    # For the creation of VM and VMSS, "--role" and "--scope" should be passed in at the same time
+    # when assigning a role to the managed identity
+    if not is_identity_assign and namespace.assign_identity is not None:
+        if (namespace.identity_scope and not namespace.identity_role) or \
+                (not namespace.identity_scope and namespace.identity_role):
```

Comment:
For [operator precedence](https://docs.python.org/3/reference/expressions.html#operator-precedence), not > and > or, so parentheses are not necessary.

Reply:
Yes, actually I know this. I put parentheses here just for clearer readability~